pytorch_lightning/plugins/training_type/tpu_spawn.py

```diff
@@ -339,10 +338,8 @@ def should_rank_save_checkpoint(self) -> bool:
     def register_plugins(cls, plugin_registry: Dict) -> None:
         plugin_registry.register("tpu_spawn_debug", cls, description="TPUSpawn Plugin with `debug` as True", debug=True)
 
-    @property
-    def checkpoint_io(self) -> CheckpointIO:
-        return self._checkpoint_io
-
-    @checkpoint_io.setter
-    def checkpoint_io(self, plugin: CheckpointIO) -> None:
-        raise MisconfigurationException("TPU Spawn Plugin currently does not support custom checkpoint plugins.")
+    @DDPSpawnPlugin.checkpoint_io.setter
+    def checkpoint_io(self, io: XLACheckpointIO) -> None:
+        if not isinstance(io, XLACheckpointIO):
+            raise MisconfigurationException(f"{self.__class__.__name__}.checkpoint_io` must be a `XLACheckpointIO`.")
+        self._checkpoint_io = io
```

Comment:
quick ques: are we forcing users to always subclass `XLACheckpointIO` and not `CheckpointIO` even if their implementation is correct?

Reply:
It's un-pythonic to have these `isinstance` checks. I prefer to remove them, however, in this PR I'm trying to keep the same functionality but with better consistency.

Before, the setter raised unconditionally, I think what it meant is to enforce a `XLACheckpointIO`

Reply:
yeah, I'd suggest by default it should use XLACheckpointIO but should allow users to be able to set a custom one too. can be updated in a separate PR.